oat-sa/extension-tao-itemqti
# Fix/ADF-1784/Blocker issues

### Related to: https://oat-sa.atlassian.net/browse/ADF-1784

### Summary

Fix a blocker issue preventing us from authoring a translation item when it contains assets.

Also, fix small styling issues:
- z-index of the resize handle
- background loader
- center position of the viewer

### Details

The legacy item runner shares a [global context](https://github.com/oat-sa/tao-item-runner-qti-fe/blob/develop/src/qtiCommonRenderer/helpers/container.js#L55). This context is used to retrieve the [DOM element attached to QTI elements](https://github.com/oat-sa/tao-item-runner-qti-fe/blob/develop/src/qtiCommonRenderer/helpers/container.js#L78-L84). This mechanism is widely used by the QTI renderers, and therefore it is not straight forward to refactor.

Instead, the creator context is force-restored once the item view is rendered.

### How to test
- checkout the branch: `git checkout -t origin/fix/ADF-1784/blocker-issues`
- also make sure to checkout the companion branches on other extensions (feat/HKD-6/integration)
- make sure to have the Feature Flags activated
```
FEATURE_FLAG_TRANSLATION_ENABLED=true
FEATURE_FLAG_UNIQUE_NUMERIC_QTI_IDENTIFIER=true
```
- make sure to have the Item Translator role assigned to your user
- have an item ready for translation, containing media or shared stimulus
- add/edit a translation: the item must be editable without error
